model/host/db.go

```diff
@@ -432,10 +432,9 @@ func ByExpiringBetween(lowerBound time.Time, upperBound time.Time) db.Q {
 func NeedsNewAgent(currentTime time.Time) db.Q {
 	cutoffTime := currentTime.Add(-MaxLCTInterval)
 	return db.Query(bson.M{
-		StatusKey:        evergreen.HostRunning,
-		StartedByKey:     evergreen.User,
-		HasContainersKey: bson.M{"$ne": true},
-		ParentIDKey:      bson.M{"$exists": false},
```

Comment:
I think that `NeedsNewAgent` will cause the provisioning_agent_deploy job to run, which in turn will not just download but also start the agent on the parent. I think we need to avoid that.

Reply:
changed the job to stop after running the setup script on parents